### ISSUE TITLE
[Bug]: Wrong position of exponent label for extended colorbar

### ISSUE THREAD

**Issue by zxdawn** [matplotlib/matplotlib#22312]
### Bug summary

The position of exponent label is wrong for the extended colorbar when the colorbar location is left or right.

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np

N = 18
state = np.random.RandomState(51423)
data = 1e5 * np.cumsum(state.rand(N, N), axis=0)
fig, ax = plt.subplots()
m = ax.pcolor(data)
fig.colorbar(m, ax=ax, extend='max', location='right')
fig.colorbar(m, ax=ax, extend='max', location='left')
```

### Actual outcome

![image](https://user-images.githubusercontent.com/30388627/150937606-8bd3ebc9-0deb-48d9-a351-9589c2e4467d.png)

### Expected outcome

Like V3.4.3

![image](https://user-images.githubusercontent.com/30388627/150937745-4ba5b7f1-009a-408b-a488-41bf11d1eb2f.png)

### Matplotlib Version

3.5.1

### Installation

conda

**Comment by jklymak** [comment]
Sorry this broke. That makes sense, and is a consequence of us changing the colorbar axes bounds to not include the extends in 3.5.  I'm not sure off the top of my head what the fix is, but hopefully will be fixed in 3.5.2 or perhaps the next bug fix release.

**Comment by jklymak** [comment]
OK -  fix wasn't too bad; confident this will be fixed 3.5.2.  

Please do let us know of any other colorbar oddities; 3.5 was a large change....
